workbench/scenarios.py

```diff
@@ -75,4 +75,12 @@ def init_scenarios():
     for class_name, cls in sorted(XBlock.load_classes(fail_silently=False)):
         add_class_scenarios(class_name, cls, fail_silently=False)
 
-init_scenarios()
+def get_scenarios():
+    """
+    Return SCENARIOS, initializing it if required.
+    """
+    if not SCENARIOS and not get_scenarios.initialized:
```

Comment:
I left that out at first but a test fails: it adds a custom XML scenario to `SCENARIOS`, then it loads the homepage, which causes `init_scenarios()` to overwrite `SCENARIOS`. Then the custom XML scenario cannot be found, and the test fails.

I also like this approach because it means that if you want to run a test like that which only involves one scenario, you can add it to `SCENARIOS` (as this test was doing), and then the expensive code to auto-detect and load all scenarios from all installed XBlocks will be skipped.